sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -386,7 +386,7 @@ internal static string GetSourcePath(Assembly assembly)
         /// <summary>
         /// Determines if the current global test mode.
         /// </summary>
-        internal static RecordedTestMode GlobalTestMode
+        public static RecordedTestMode GlobalTestMode
```

Comment:
@pakrym @JoshLove-msft Could you advise better strategy here? I'm in a need of global fixture that has both tear down and setup capability and is only active in recording and live mode, not playback (see `ManagedDiskFixture` below). 

Initially, I was exploring "one time setup" per test but that turned out to be not enough isolation for that case and I want to avoid spinning up a VM with managed disk per test class...

Options that comes to my mind:
1. Keep it as is, just publish the `GlobalTestMode` property
2. Make `GlobalTestMode` internal protected so derived test envs can make use of it
3. Default `TestEnvironment.Mode` to `GlobalTestMode`, (so that I can construct test env instance in my fixture and it gets intilized with some `Mode` instead of relying on `RecoredTestBase` to initialize it)
4. Make the fixture extend `RecoredTestBase` - I don't like this, it's a hack and abuse of inheritance, the fixture isn't a test.
5. Create base class `LiveSetupFixture` in test framework package that would have `[OneTimeSetup]` and `[OneTimeTeardown]` calling into abstract methods that implementer has to override (adapter?).
6. Try creating `[LiveSetupFixture]` attribute, by bending [this](https://github.com/nunit/nunit/blob/master/src/NUnitFramework/framework/Attributes/SetUpFixtureAttribute.cs) and [this](https://github.com/nunit/nunit/blob/master/src/NUnitFramework/framework/Internal/Tests/SetUpFixture.cs) somehow.

Reply:
Discussed offline and we've chosen to pursue option 3.

Reply:
Well, option 3 was an uphill battle and was abandoned in favor of some variation of option 5.